sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -83,6 +85,19 @@ class ProjectOverviewView extends VerticalLayout implements HasHotbar, HasTitle
 
     }
 
+    /**
+     *
+     * @param consumer
+     */
+    public void onProjectDoubleClick(Consumer<ProjectSummary> consumer){
+         projectGrid.addItemClickListener({
+             //if grid.getEditor().setEnabled(true) is enabled this will not work anymore!
+             if(it.mouseEventDetails.isDoubleClick()){
+                 if(viewModel.selectedProject) consumer.accept(viewModel.selectedProject)
```

Comment:
The null check here can be circumvented as you only want to perform an action in case a value is present. This adds dynamic evaluation and makes the code more readable imo
```suggestion
                 Optional.ofNullable(viewModel.selectedProject).ifPresent(consumer)
```

Reply:
I actually think it is not more readable since the null check is the condition for the consumer to be called. If its not super important for you to have that here I would much prefer to keep it as is. I think this is a subjective thing and does not change functionality.

Reply:
When the `if` statement is used it is a good style to put the curly brackets around it.
The solution would look like this
```
if (viewModel.selectedProject) {
    consumer.accept(viewModel.selectedProject)
}
```
compared to
```
Optional.ofNullable(viewModel.selectedProject).ifPresent(consumer)
```

It is your choice which of those options you choose. However, I would kindly ask you to add the curly brackets indicating the scope of the block more clearly.